src/Avalonia.Base/Rendering/Composition/Server/ServerCompositionVisual.cs

```diff
@@ -128,11 +132,6 @@ public virtual UpdateResult Update(ServerCompositionTarget root, Matrix parentVi
 
             var wasVisible = IsVisibleInFrame;
 
-            if(Parent != null)
-            {
-                RenderOptions = RenderOptions.MergeWith(Parent.RenderOptions);
-            }
```

Comment:
Note for myself: MergeWith is executed from PushRenderOptions (not quite obvious at first what this PR does)

Reply:
Direct setter is removed completely in one of the next PRs, maybe it's worth to move those changes to this PR instead